modules/gapi/test/common/gapi_imgproc_tests_common.hpp

```diff
@@ -14,6 +14,105 @@
 
 namespace opencv_test
 {
+// Draw random ellipses on given mat of given size and type
+static void initMatForFindingContours(cv::Mat& mat, const cv::Size& sz, const int type)
+{
+    cv::RNG& rng = theRNG();
+    mat = cv::Mat(sz, type, cv::Scalar::all(0));
+    size_t numEllipses = rng.uniform(1, 10);
+
+    for( size_t i = 0; i < numEllipses; i++ )
+    {
+        cv::Point center;
+        cv::Size  axes;
+        center.x     = rng.uniform(0, sz.width);
+        center.y     = rng.uniform(0, sz.height);
+        axes.width   = rng.uniform(2, sz.width);
+        axes.height  = rng.uniform(2, sz.height);
+        int color    = rng.uniform(1, 256);
+        double angle = rng.uniform(0., 180.);
+        cv::ellipse(mat, center, axes, angle, 0., 360., color, 1, FILLED);
+    }
+}
+
+enum WithHierarchy {NO_HIERARCHY, WITH_HIERARCHY};
+
+template<WithHierarchy withHierarchy = NO_HIERARCHY>
+cv::GComputation findContoursTestGAPI(const cv::Mat& in, const cv::RetrievalModes mode,
+                                      const cv::ContourApproximationModes method,
+                                      cv::GCompileArgs&& args,
+                                      std::vector<std::vector<cv::Point>>& out_cnts_gapi,
+                                      std::vector<cv::Vec4i>&,
```

Comment:
`std::vector<cv::Vec4i>& /*out_hier_gapi*/,`

for clarity

Reply:
If you think it is necessary. Applied